### PR TITLE
Automated Transfer compatibility

### PR DESCRIPTION
**Pull request by kwonye**
Resolves the issue of Automated Transfer (AT) sites constantly asking for the password.

What we do is store `is_automated_transfer` to the Blog object and disable seeing it from the app. Here are the multiple cases:

1. The AT site is the one-and-only site for the user. There is a page telling the user their site is not compatible with this version of the app. They can then add a new site.
2. The AT site is one of multiple sites they have. The page then says the same warning, but allows the user to select a different site from the list.

cc @catehstn @rralian

### CODE REVIEW

**Comment by kwonye** [comment]
@maxme I'm running into a problem where I the accounts table won't add the new `isAutomatedTransfer` column. Can you look at the code and see if you can find what I did wrong please? Thanks!

File: WordPress/src/main/java/org/wordpress/android/ui/main/MySiteFragment.java [hunk 1]
```diff
@@ -333,17 +335,28 @@ private void refreshBlogDetails(@Nullable final Blog blog) {
...
+            showNoSitesView();
+
+            return;
+        } else if (blog.getAutomatedTransfer()) {
+            showNoSitesView();
+
+            int numBlogs = WordPress.wpDB.getNumBlogs();
+            WPTextView title = (WPTextView) getView().findViewById(R.id.my_site_no_site_title);
+            WPTextView prompt = (WPTextView) getView().findViewById(R.id.my_site_no_site_prompt);
+            title.setText("Your site is not supported in this version of the app. Please update your app in the Play Store");
```
Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/ui/main/MySiteFragment.java]: I know it's a temporary hack, but i18n matters (for this string and strings below).

**Comment by maxme** [comment]
@kwonye migration works (I double checked by pulling and inspecting the DB), and the migration code looks good, but the patch doesn't work:

- [ ] in 6.7, be logged in a wpcom user with multiple sites (no AT), install this version and go to My Site, you'll get the "no sites" screen.

**Comment by maxme** [comment]
Also, I don't think we need this kind of patch at all, can't we just force pull the site list when the app starts? AT sites will be automatically hidden because they will be marked as Jetpack sites.

**Comment by maxme** [comment]
Note: updated base to `release/6.8`

File: WordPress/src/main/java/org/wordpress/android/ui/main/MySiteFragment.java (hunk 1 above)
Comment (kwonye) [inline WordPress/src/main/java/org/wordpress/android/ui/main/MySiteFragment.java]: You're totally right, sorry. In my haste I forgot to factor that in. Will update.

**Comment by kwonye** [comment]
>  in 6.7, be logged in a wpcom user with multiple sites (no AT), install this version and go to My Site, you'll get the "no sites" screen.

@maxme does it work for the single-AT site only?

> Also, I don't think we need this kind of patch at all, can't we just force pull the site list when the app starts? AT sites will be automatically hidden because they will be marked as Jetpack sites.

The reason for this patch is to account for the single-site AT situation. Even though the site is "hidden", the user can access the "My Site" fragment because it gets the current site. Because of that, there is a pop up for incorrect credentials when the app tries to fetch information about the site. Try the following to see what I mean:

1. Sign in to a site with a single AT site
2. Go to the My Sites tab
3. Force quit the app
4. Come back to the app. You'll see the "Incorrect credentials" pop up

**Comment by maxme** [comment]
> Even though the site is "hidden", the user can access the "My Site" fragment because it gets the current site. Because of that, there is a pop up for incorrect credentials when the app tries to fetch information about the site. Try the following to see what I mean:

So I think if the selected site is hidden, we should just reset the selected site. This should show the first site if the user has other sites or show the drake empty screen if it was his only site.

**Comment by kwonye** [comment]
> So I think if the selected site is hidden, we should just reset the selected site. This should show the first site if the user has other sites or show the drake empty screen if it was his only site.

Good call, let's do this.

File: WordPress/src/main/java/org/wordpress/android/models/Blog.java [hunk 2]
```diff
@@ -367,6 +369,14 @@ public String getBlogOptions() {
         return blogOptions;
     }
 
+    public void setAutomatedTransfer(boolean automatedTransfer) {
+        isAutomatedTransfer = automatedTransfer;
+    }
+
+    public boolean getAutomatedTransfer() {
```
Comment (mzorz) [inline WordPress/src/main/java/org/wordpress/android/models/Blog.java]: I think this method would be better named `isAutomatedTransfer()`

File: WordPress/src/main/java/org/wordpress/android/ui/main/MySiteFragment.java [hunk 3]
```diff
@@ -327,7 +327,11 @@ public float getInterpolation(float t) {
         }
     }
 
-    private void refreshBlogDetails(@Nullable final Blog blog) {
+    private void refreshBlogDetails(@Nullable Blog blog) {
+        if (blog != null && blog.getAutomatedTransfer()) {
+            mBlogLocalId = WordPress.wpDB.getFirstVisibleAndNonAutomatedTransferBlogId();
+            WordPress.setCurrentBlog(mBlogLocalId);
```
Comment (mzorz) [inline WordPress/src/main/java/org/wordpress/android/ui/main/MySiteFragment.java]: I still have to try this, but I think here if there is no other blog (i.e. the only existing blog is an automated transfer blog), then we're setting the current blog but then the UI is setup some lines below within this method, with the `blog` object populated. If the intention is to hide these sites, then we should add `blog = null;` right after `setCurrentBlog` in this `if` statement.

Comment (mzorz) [inline WordPress/src/main/java/org/wordpress/android/ui/main/MySiteFragment.java]: tried and verified it was an issue - addressed this one here acbc1f4

File: WordPress/src/main/java/org/wordpress/android/ui/accounts/BlogUtils.java [hunk 4]
```diff
@@ -161,7 +161,7 @@ public static boolean addOrUpdateBlog(String blogName, String xmlRpcUrl, String
                     blog.setCapabilities(capabilities);
                     blogUpdated = true;
                 }
-                if (isAutomatedTransfer) {
+                if (isAutomatedTransfer != blog.getAutomatedTransfer()) {
```
Comment (kwonye) [inline WordPress/src/main/java/org/wordpress/android/ui/accounts/BlogUtils.java]: Good change, this should future-proof this a tiny bit if there's ever a case where we can switch a site back from AT in the future.

**Comment by kwonye** [comment]
Great work, @mzorz! You've fixed every issue I was running across.

It's a :shipit: from me, but since I technically did the last commit and started this PR, I'll have you give the final word and merge it. 👌

**Comment by mzorz** [comment]
Great! Performed all these tests successfully:

### Test 1:
- having a list with 2 .com sites, and no AT sites, select one normal .com site from your list in the app, make it AT in Calypso, wait for 15 minutes (or change the device time 15 minutes ahead) and bring the app back to the foreground. This will make the blog list sync. 
Verify the app should select the first non AT site if only one available

### Test 2:
- after performing TEST 1, you should end up having just one .com site. Now remove your user as an admin of this site on Calypso (of course another user should remain an admin of the site). Wait for 15 minutes (or change the device time 15 minutes ahead) and bring the app back to the foreground.
Verify the app should show the drake screen as no other sites found.

### Test 3:
- having a list with 1 (one) .com sites, and no AT sites, select the .com site from your list, make it AT, wait for 15 minutes (or change the device time 15 minutes ahead) and bring the app back to the foreground. This will make the blog list sync. 
Verify the app should show the drake screen as no other sites found.

### Test 4:
- Having an AT site, set up the Android app for first time, adding this site. How would the user (or the app, if it makes sense) know that this is a self hosted site?
(question, what if I have an AT site, but never used the app, then try to login to the app. What should be expected to happen?)
IT SHOULD LET YOU LOGIN BUT THEN SHOW NO SITES IN THE LIST.

### Test 5:
- Having 2 or more .com sites, make all of them AT, and no sites should be shown on the list.

### Test 6:
- Having the old app, test an AT site and check it doesn’t work correctly (see Plans or Stats).
Now update the app with this branch, and check it falls back to the first available visible site.

**Comment by mzorz** [comment]
:shipit:

**Pull request merged**